acceptance/tests/load_libfacter.rb

```diff
@@ -1,31 +1,66 @@
-test_name 'Ruby can load libfacter'
+# Verify that we can load the facter the way that mco does based on the platform
+test_name 'C100161: Ruby can load libfacter without raising an error' do
+  tag 'risk:high'
 
-agents.each do |agent|
-  on agent, "env PATH=\"#{agent['privatebindir']}:${PATH}\" which ruby"
-  ruby_path = stdout.chomp
+  require 'puppet/acceptance/common_utils'
+  extend Puppet::Acceptance::CommandUtils
 
-  root_dir = ruby_path
-  while File.basename(root_dir).downcase != 'puppet'
-    new_root_dir = File.dirname(root_dir)
-    if new_root_dir == root_dir
-      break
-    else
-      root_dir = new_root_dir
+  def puppet_ruby_path_to_puppet_install_dir(puppet_ruby_path)
+    # find the "puppet" directory which should be the root of the install
+    puppet_dir = puppet_ruby_path
+    while File.basename(puppet_dir).downcase != 'puppet'
+      new_puppet_dir = File.dirname(puppet_dir)
+      if new_puppet_dir == puppet_dir
+        break
+      else
+        puppet_dir = new_puppet_dir
+      end
     end
+    puppet_dir
   end
 
-  if agent.platform.variant == 'windows'
-    root_dir = on(agent, "cygpath -w '#{root_dir}'").stdout.chomp
-    facter_loader = "#{root_dir}/facter/lib/facter.rb"
-    path = '/cygdrive/c/Windows/system32:/cygdrive/c/Windows'
-  else
-    facter_loader = "#{root_dir}/lib/ruby/vendor_ruby/facter.rb"
-    path = '/usr/local/bin:/bin:/usr/bin'
-  end
+  agents.each do |agent|
+    # create a ruby program that will add a fact through Facter
+    fact_content = <<-EOM
+    require 'facter'
+    Facter.add('facter_loaded') do
+      setcode do
+        'FACTER_LOADED'
+      end
+    end
 
-  # Ensure privatebindir comes first so we get the correct Ruby version.
-  on agent, "env PATH=#{path} \"#{ruby_path}\" \"#{facter_loader}\"" do
-    assert_empty stdout
-    assert_empty stderr
+    # print a value so that we know that facter loaded and is working
+    puts Facter.value('facter_loaded')
+    exit 0
+    EOM
+
+    fact_dir = agent.tmpdir('mco_test')
+    fact_program = File.join(fact_dir, 'loading_facter.rb')
+    create_remote_file(agent, fact_program, fact_content)
+
+    teardown do
+      on(agent, "rm -rf '#{fact_dir}'")
+    end
+
+    if agent['platform'] =~ /windows/
+      # on Windows we have to figure out where facter.rb is so we can include the path
+      # figure out the root of the Puppet installation
+      puppet_ruby_path = on(agent, "env PATH=\"#{agent['privatebindir']}:${PATH}\" which ruby").stdout.chomp
+      cygwin_puppet_root = puppet_ruby_path_to_puppet_install_dir(puppet_ruby_path)
+      puppet_root = on(agent, "cygpath -w '#{cygwin_puppet_root}'").stdout.chomp
```

Comment:
@ericldelaney it is unfortunate that we need all of this jiggery-pokery just to get the root install directory of `puppet-agent`. Can you file a Beaker ticket to have such a facility added to Beaker core?